ui/app/src/components/DeletePluginDialog/DeletePluginDialogContainer.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2007-2021 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import { DeletePluginDialogContainerProps } from './utils';
+import { useActiveSiteId } from '../../utils/hooks/useActiveSiteId';
+import { useMemo, useRef } from 'react';
+import { createResource } from '../../utils/resource';
+import { deleteMarketplacePlugin, fetchMarketplacePluginUsage } from '../../services/marketplace';
+import Suspencified from '../SystemStatus/Suspencified';
+import { DeletePluginDialogBody } from './DeletePluginDialogBody';
+import { useDispatch } from 'react-redux';
+import { showSystemNotification } from '../../state/actions/system';
+import { showErrorDialog } from '../../state/reducers/dialogs/error';
+import { defineMessages, useIntl } from 'react-intl';
+
+const messages = defineMessages({
+  pluginDeleted: {
+    id: 'pluginManagement.pluginDeleted',
+    defaultMessage: 'Plugin deleted'
+  }
+});
+
+export function DeletePluginDialogContainer(props: DeletePluginDialogContainerProps) {
+  const { onClose, pluginId, onComplete, isSubmitting, onSubmittingAndOrPendingChange } = props;
+  const site = useActiveSiteId();
+  const dispatch = useDispatch();
+  const { formatMessage } = useIntl();
+  const callbacksRef = useRef({ onSubmittingAndOrPendingChange });
+  callbacksRef.current.onSubmittingAndOrPendingChange = onSubmittingAndOrPendingChange;
```

Comment:
`import useRef from '../../utils/hooks/useUpdateRefs';` instead of react's

```suggestion
  const callbacksRef = useRef({ onSubmittingAndOrPendingChange });
```